### PR TITLE
fixes bug 1316407 - Create a suite of e2e tests for the API

### PR DESCRIPTION
**Pull request by peterbe**
CC @m8ttyB

### CODE REVIEW

**Comment by m8ttyB** [comment]
Since these are read-only tests, add `@pytest.mark.nondestructive` to each test. Other than that tests look good.

File: e2e-tests/tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from collections import defaultdict
+
+import requests
+import pytest  # noqa
+
+
+class TestAPI:
+
```
Comment (m8ttyB) [inline e2e-tests/tests/test_api.py]: add `@pytest.mark.nondestructive` to each test method

**Comment by peterbe** [comment]
Done.

File: e2e-tests/tests/test_api.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            })
+            assert response.status_code == 200
+            assert response.json()['hits']
+
+    @pytest.mark.nondestructive
+    def test_supersearch_fields(self, base_url):
+        response = requests.get(base_url + '/api/SuperSearchFields/')
+        assert response.status_code == 200
+        fields = response.json()
+        assert fields['uuid'], fields
```
Comment (willkg) [inline e2e-tests/tests/test_api.py]: I don't understand this line. pytest generally produces good assertion error messages. Seems like this is testing to see if `fields['uuid']` is truthy, but I suspect it's really just checking to see if `'uuid'` is in fields.

I'm going to tweak this in a followup commit.

**Comment by willkg** [comment]
This looks good to me.

@adngdb Can you check to see if the tests are helpful and/or if they're missing anything? If it's good, we should land it.

**Pull request merged**